apps/mobile/src/features/approver/layouts/base-stx-message-approver.layout.tsx

```diff
@@ -59,14 +65,25 @@ export function BaseStxMessageApproverLayout({
         )}
       </Approver.Container>
       <Approver.Footer>
-        <Approver.Actions>
-          <Button variant="outline" flex={1} onPress={onCloseApprover}>
-            {t`Deny`}
-          </Button>
-          <Button flex={1} onPress={onApprove}>
-            {t`Approve`}
-          </Button>
-        </Approver.Actions>
+        <ApproverButtons
+          ref={buttonsRef}
+          onBack={onCloseApprover}
+          onApprove={() => {
+            buttonsRef.current?.onBroadcast();
```

Comment:
I don't really follow why these events are handled this way? These could be triggered internally no?